### PR TITLE
Rename error group fingerprint to error group

### PR DESCRIPTION
**Pull request by nrcventura**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description

Updates the error group fingerprint attribute name, definition, and enum to match the names specified in #1399. 

# Author Checklist
- ~~[ ] Unit tests, Integration tests, and Unbounded tests completed~~ Just a rename.
- ~~[ ] Performance testing completed with satisfactory results (if required)~~ Just a rename
- ~~[ ] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated~~ Just a rename

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

**Review by tippmar-nr — APPROVED** [review]

File: src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs [hunk 1]
```diff
@@ -632,7 +632,7 @@ void work()
             {
                 InternalApi.ErrorFingerprintingCallback(callback);
             }
-            TryInvoke(work, apiName, ApiMethod.ErrorFingerprintingCallback);
+            TryInvoke(work, apiName, ApiMethod.ErrorGroupCallback);
```
Comment (chynesNR) [inline src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs:635]: Could we have `Set` in the name, so it's clear what it's doing? This makes it seem like we're invoking the callback here.

File: src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs [hunk 2]
```diff
@@ -632,7 +632,7 @@ void work()
             {
                 InternalApi.ErrorFingerprintingCallback(callback);
             }
-            TryInvoke(work, apiName, ApiMethod.ErrorFingerprintingCallback);
+            TryInvoke(work, apiName, ApiMethod.SetErrorGroupCallback);
```
Comment (jaffinito) [inline src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs:635]: In the most recent spec update, the callback name is still ErrorGroupCallback (error_group_callback), so I don't think we should call this "SetErrorGroupCallback".

File: src/Agent/NewRelic/Agent/Core/NewRelic.Agent.Core.Metric/ApiSupportabilityMetricCounters.cs [hunk 3]
```diff
@@ -38,7 +38,7 @@ public enum ApiMethod
         InsertDistributedTraceHeaders = 24,
         AcceptDistributedTraceHeaders = 25,
         SpanSetName = 26,
-        ErrorFingerprintingCallback = 27,
+        SetErrorGroupCallback = 27,
```
Comment (jaffinito) [inline src/Agent/NewRelic/Agent/Core/NewRelic.Agent.Core.Metric/ApiSupportabilityMetricCounters.cs:41]: Same a previous comment.

File: src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs (hunk 1 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs]: The spec currently doesn't have the set verb.

File: src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs (hunk 2 above)
Comment (nrcventura) [reply src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs:635]: Is the intention of the API to be used as `Agent.ErrorGroupCallback(MyType.MyCallback)` or `Agent.ErrorGroupCallback = MyType.MyCallback`? Given the name, I think a property syntax makes the most sense, in which case the `Set` is implied. The method form is kind of confusing to me.

Comment (chynesNR) [reply src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs:635]: Ah, I took that to mean the name of the parameter, not the API call. It does look like the spec typically calls out the function name. I guess I should have raised this then. :)

File: src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs (hunk 1 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs]: We discussed this in pairing and decided to go with Set...

File: src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs (hunk 2 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs:635]: We discussed this in pairing and decided to go with Set...

File: src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs (hunk 1 above)
Comment (nrcventura) [reply src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs]: I plan on updating the API itself (and not just the enum) with a separate PR.

**Review by tippmar-nr — APPROVED** [review]

**Pull request merged**